Sources/ArcGISToolkit/Components/FloorFilter/SiteAndFacilitySelector.swift

```diff
@@ -130,14 +131,19 @@ struct SiteAndFacilitySelector: View {
                         facilities: site.facilities,
                         isHidden: isHidden
                     )
-                    .onDisappear {
-                        shouldAutoSelect = false
-                    }
+                    .navigationBarBackButtonHidden(true)
+                    .navigationBarItems(
+                        leading: Button {
```

Comment:
Because SwiftIUI doesn't provide a way to get a callback when the back arrow in a navigation view is pressed, we hide the default back button and provide our own which gives us this functionality.